include/fmt/format.h

```diff
@@ -2854,26 +2867,39 @@ void basic_writer<Range>::write_double(T value, const format_specs& spec) {
     basic_writer& writer;
     format_specs spec;
     char sign;
+    bool as_percentage;
     void operator()(const char* str) const {
-      writer.write_padded(spec, inf_or_nan_writer{sign, str});
+      writer.write_padded(spec, inf_or_nan_writer{sign, as_percentage, str});
     }
-  } write_inf_or_nan = {*this, spec, sign};
+  } write_inf_or_nan = {*this, spec, sign, handler.as_percentage};
 
   // Format NaN and ininity ourselves because sprintf's output is not consistent
   // across platforms.
-  if (internal::fputil::isnotanumber(value))
-    return write_inf_or_nan(handler.upper ? "NAN" : "nan");
-  if (internal::fputil::isinfinity(value))
-    return write_inf_or_nan(handler.upper ? "INF" : "inf");
+  if (internal::fputil::isnotanumber(value)) {
+    write_inf_or_nan(handler.upper ? "NAN" : "nan");
+    return;
+  }
+  if (internal::fputil::isinfinity(value)) {
+    write_inf_or_nan(handler.upper ? "INF" : "inf");
+    return;
+  }
 
   memory_buffer buffer;
   int exp = 0;
   int precision = spec.has_precision() || !spec.type ? spec.precision : 6;
+
+  if (handler.as_percentage) value *= 100.;
+
   bool use_grisu = fmt::internal::use_grisu<T>() &&
                    (!spec.type || handler.fixed) &&
                    internal::grisu2_format(static_cast<double>(value), buffer,
                                            precision, handler.fixed, exp);
   if (!use_grisu) internal::sprintf_format(value, buffer, spec);
+
+  if (handler.as_percentage) {
+    buffer.push_back('%');
+    --exp;  // Adjust decimal place position.
```

Comment:
Shouldn't this be -= 2?

Reply:
The decimal place position is shifted by one place because only one character, the percent sign, is added.